pg_search/src/index/directory/channel.rs

```diff
@@ -288,8 +301,9 @@ impl ChannelRequestHandler {
             self.process_message(message)?;
         }
 
-        let func: Action = Box::new(move || Box::new(action_func()));
-        self.action.0.send(func)?;
+        let boxed_func: Action<'me> = Box::new(move || Box::new(func()));
+        let boxed_func: Action<'static> = unsafe { std::mem::transmute(boxed_func) };
```

Comment:
I should note that what we're doing here is transmuting the lifetime of the `FnOnce() -> T` argument `func` from `'me` (meaning it's assumed to borrow from `'self`) to '`static'.  I'm p sure this is safe because despite the closure getting passed to a background thread, we actually wait on it through the internal `self.action` channel and `self.reply` channels.